pkg/playback/http_playback.go

```diff
@@ -262,9 +271,24 @@ func (rr *Server) InitializeServer(address string) *http.Server {
 	return server
 }
 
+// OnSwitchMode blocks on the switchModeChan and returns the value it receives
+// func (rr *Server) OnSwitchMode(send chan<- string) {
+// 	mode := <-rr.switchModeChan
+// 	send <- mode
+// }
+
+// OnSwitchMode blocks on the switchModeChan and returns the value it receives
+func (rr *Server) OnSwitchMode(f func(string)) {
+	go func() {
```

Comment:
also not sure what the best practice in go is.

calling `go func() ...` inside a function and simply calling this function inline `rr.OnSwitchMode(...)` 

or calling the logic inline without setting up a goroutine inside the function, and having the caller setup a go routing (`go rr.OnSwitchMode(...)`)?

Reply:
This works!

The other thing you can do is have `OnSwitchMode` put the "handler" (`f`) on the struct, and then start a goroutine up from `NewServer` that reads the channel and fires the handler if set. That might have better behavior w.r.t. if `OnSwitchMode` were to be called multiple times.

Reply:
I think this also only fires the handler once, but previously it would call it multiple times?

Reply:
Adding the handler on the struct would imply you want an array of handlers in case `OnSwitchMode` is called multiple times, right?

I think this solution is probably good enough for now - I don't love it and hopefully will be able to cleanup later on when we refactor _everything_ !